tests/TestCli.cpp

```diff
@@ -433,6 +442,163 @@ void TestCli::testAnalyze()
     QCOMPARE(m_stderr->readAll(), QByteArray());
 }
 
+void TestCli::testAttachmentExport()
+{
+    TemporaryFile out;
+    QVERIFY(out.open(QIODevice::ReadWrite));
+
+    FILE* fout = fdopen(out.handle(), "w");
+    QVERIFY(fout);
+
+    AttachmentExport attachmentExportCmd(fout);
+    QVERIFY(!attachmentExportCmd.name.isEmpty());
+    QVERIFY(attachmentExportCmd.getDescriptionLine().contains(attachmentExportCmd.name));
+
+    // Try exporting an attachment of a non-existent entry
+    setInput("a");
+    execCmd(attachmentExportCmd,
+            {"attachment-export", m_dbFile->fileName(), "invalid_entry_path", "invalid_attachment_name"});
+    m_stderr->readLine(); // skip password prompt
+    QCOMPARE(m_stderr->readAll(), QByteArray("Could not find entry with path invalid_entry_path.\n"));
+    QCOMPARE(m_stdout->readAll(), QByteArray());
+
+    // Try exporting a non-existent attachment
+    setInput("a");
+    execCmd(attachmentExportCmd,
+            {"attachment-export", m_dbFile->fileName(), "/Sample Entry", "invalid_attachment_name"});
+    m_stderr->readLine(); // skip password prompt
+    QCOMPARE(m_stderr->readAll(), QByteArray("Could not find attachment with name invalid_attachment_name.\n"));
+    QCOMPARE(m_stdout->readAll(), QByteArray());
+
+    // Export an existing attachment
+    setInput("a");
+    execCmd(attachmentExportCmd, {"attachment-export", m_dbFile->fileName(), "/Sample Entry", "Sample attachment.txt"});
+    m_stderr->readLine(); // skip password prompt
+    QCOMPARE(m_stderr->readAll(), QByteArray());
+
+    fclose(fout);
+
+    // Open a second QFile here as out does not notice the changes made to fout, resulting in stdout.readAll()
+    // returning an empty string if it were to be initialized using stdout(&out).
+    QFile check(out.fileName());
+    QVERIFY(check.open(QIODevice::ReadOnly));
+    QCOMPARE(check.readAll(), QByteArray("Sample content\n"));
+}
+
+void TestCli::testAttachmentImport()
+{
+    AttachmentImport attachmentImportCmd;
+    QVERIFY(!attachmentImportCmd.name.isEmpty());
+    QVERIFY(attachmentImportCmd.getDescriptionLine().contains(attachmentImportCmd.name));
+
+    const QString attachment = QString(KEEPASSX_TEST_DATA_DIR).append("/Attachment.txt");
```

Comment:
@andreblanke `attachmentPath` maybe? otherwise it's not clear if it's the path of the content of the attachment.